spec/models/spree/gateway/amazon_spec.rb

```diff
@@ -14,20 +25,19 @@
 
       expect(mws).to have_received(:refund).with("CAPTURE_ID", payment.number, 30.0, "USD")
     end
-  end
 
-  describe '#authorize' do
-    let(:payment_method) { Spree::Gateway::Amazon.create!(name: 'Amazon', preferred_test_mode: true) }
-    let(:order) { create(:order_with_line_items, state: 'delivery') }
-    let(:payment_source) { Spree::AmazonTransaction.create!(order_id: order.id, order_reference: 'REFERENCE') }
-    let!(:payment) do
-      create(:payment,
-             order: order,
-             payment_method: payment_method,
-             source: payment_source,
-             amount: order.total)
+    let!(:refund) { create(:refund, payment: payment, amount: payment.amount) }
+    it 'succeeds' do
+      response = build_mws_refund_response(state: 'Pending', total: order.total)
+      expect(payment_method.send(:load_amazon_mws, 'REFERENCE')).to receive(:refund).and_return(response)
```

Comment:
Yeah that was my bad, stubbing a private method is not very good, so we should change it
to the class.